### ISSUE TITLE
Const trait impl is marked as a syntax error

### ISSUE THREAD

**Issue by mbartlett21** [rust-analyzer/rust-analyzer#7714]
The `const` trait impl below is marked as a syntax error: expected type.

Rust-analyser should instead parse this, ignoring the `const` qualifier

```rust
impl const Trait for Something {}
//   ^^^^^ Syntax Error: expected type
```

**Comment by Veykril** [comment]
Are you on the latest version? This has been implemented a few weeks ago #7338

**Comment by mbartlett21** [comment]
Updating fixed it :)
